src/python/grapl_analyzerlib/grapl_analyzerlib/queryable.py

```diff
@@ -169,7 +165,46 @@ def set_property_filters(self, property_name: str, filters: List[List["Cmp"]]):
     def set_neighbor_filters(self, edge_name: str, filters: EdgeFilter[Q]):
         self._edge_filters[edge_name].extend(filters)
 
+    def query(self, graph_client, first: int) -> List[V]:
+        var_alloc, query = gen_query(self, "q0", first=first)
+
+        variables = {v: k for k, v in var_alloc.allocated.items()}
+        txn = graph_client.txn(read_only=True)
+
+        try:
+            qres = json.loads(txn.query(query, variables=variables).json)
+        finally:
+            txn.discard()
```

Comment:
this implies to me that we should introduce a `with graph_client.txn_context(...):` context manager, no?

Reply:
Yeah that'd be way cleaner.